clang/lib/Driver/ToolChains/Flang.cpp

```diff
@@ -345,7 +345,13 @@ void Flang::AddAMDGPUTargetArgs(const ArgList &Args,
 
   // Check ROCm path if specified
   const ToolChain &TC = getToolChain();
-  TC.getDeviceLibs(Args);
+  std::string HIPVersion;
+  llvm::raw_string_ostream HIPInfo(HIPVersion);
+  TC.printVerboseInfo(HIPInfo);
+  llvm::StringRef HIPInfoStrRef(HIPInfo.str());
+  if (!HIPInfoStrRef.contains("Found HIP installation") &&
```

Comment:
Clang reports an error if `--rocm-path` points to an invalid directory. I need to do a similar check for Flang.

The ROCm toolchain checks rocm-path within function `RocmInstallationDetector::checkCommonBitcodeLibs`. This function is called inside `ROCMToolChain::addClangTargetOptions` or `ROCMToolChain::getCommonDeviceLibNames`. `ROCMToolChain::getCommonDeviceLibNames` is called by `AMDGPUOpenMPToolChain::getDeviceLibs`. I decided not to call  `ROCMToolChain::addClangTargetOptions` because Flang does not support all Clang options. That's why I initially decided to call `AMDGPUOpenMPToolChain::getDeviceLibs` to check the ROCm path. The second (current) approach is a workaround to emit an error if rocm-path is specified incorrectly.

Reply:
Do you know where clang emits that diagnostic? I'd assume it would be done during ROCm detection.

Reply:
test command:
`clang -fopenmp --offload-arch=gfx90a test.c`

Function call:
``` 
clang/lib/Driver/ToolChains/Clang.cpp : void Clang::ConstructJob
TC.addClangTargetOptions(Args, CmdArgs, JA.getOffloadingDeviceKind());
                                  |
                                  V
clang/lib/Driver/ToolChains/AMDGPUOpenMP.cpp: void AMDGPUOpenMPToolChain::addClangTargetOptions
   for (auto BCFile : getDeviceLibs(DriverArgs)) {
                                  |
                                  V
 clang/lib/Driver/ToolChains/AMDGPUOpenMP.cpp: AMDGPUOpenMPToolChain::getDeviceLibs (
                                  |
                                  V
   if (!RocmInstallation->hasDeviceLibrary()) {
        getDriver().Diag(diag::err_drv_no_rocm_device_lib) << 0;
        return {};
  }
  ```

Reply:
Do we use these libraries at all yet?

Reply:
No, we do not use them in Flang